tools/cjxl_ng_main.cc

```diff
@@ -427,467 +413,437 @@ int main(int argc, char** argv) {
-      JxlEncoderFrameSettingsSetOption(jxl_encoder_frame_settings,
-                                       JXL_ENC_FRAME_SETTING_PROGRESSIVE_AC,
-                                       FLAGS_progressive_ac);
-    }
-    if (responsive_set) {
-      JxlEncoderFrameSettingsSetOption(jxl_encoder_frame_settings,
-                                       JXL_ENC_FRAME_SETTING_RESPONSIVE,
-                                       responsive);
-    }
-    if (qprogressive_ac_set) {
-      JxlEncoderFrameSettingsSetOption(jxl_encoder_frame_settings,
-                                       JXL_ENC_FRAME_SETTING_QPROGRESSIVE_AC,
-                                       qprogressive_ac);
-    }
-  }
-  // Modular mode related
-  {
-    bool modular_group_size_set =
-        !gflags::GetCommandLineFlagInfoOrDie("modular_group_size").is_default;
-    bool modular_predictor_set =
-        !gflags::GetCommandLineFlagInfoOrDie("modular_predictor").is_default;
-    bool modular_colorspace_set =
-        !gflags::GetCommandLineFlagInfoOrDie("modular_colorspace").is_default;
-    bool modular_ma_tree_learning_percent_set =
-        !gflags::GetCommandLineFlagInfoOrDie("modular_ma_tree_learning_percent")
-             .is_default;
-    bool modular_nb_prev_channels_set =
-        !gflags::GetCommandLineFlagInfoOrDie("modular_nb_prev_channels")
-             .is_default;
-    bool modular_lossy_palette_set =
-        !gflags::GetCommandLineFlagInfoOrDie("modular_lossy_palette")
-             .is_default;
-    bool modular_palette_colors_set =
-        !gflags::GetCommandLineFlagInfoOrDie("modular_palette_colors")
-             .is_default;
-    bool modular_channel_colors_global_percent_set =
-        !gflags::GetCommandLineFlagInfoOrDie(
-             "modular_channel_colors_global_percent")
-             .is_default;
-    bool modular_channel_colors_group_percent_set =
-        !gflags::GetCommandLineFlagInfoOrDie(
-             "modular_channel_colors_group_percent")
-             .is_default;
-
-    if (modular_group_size_set) {
-      if (!(FLAGS_modular_group_size == -1 ||
-            (0 <= FLAGS_modular_group_size && FLAGS_modular_group_size <= 3))) {
-        std::cerr << "Invalid --modular_group_size: "
-                  << FLAGS_modular_group_size << std::endl;
-        return EXIT_FAILURE;
-      }
-      JxlEncoderFrameSettingsSetOption(jxl_encoder_frame_settings,
-                                       JXL_ENC_FRAME_SETTING_MODULAR_GROUP_SIZE,
-                                       FLAGS_modular_group_size);
-    }
-    if (modular_predictor_set) {
-      if (!(0 <= FLAGS_modular_predictor && FLAGS_modular_predictor <= 3)) {
-        std::cerr << "Invalid --modular_predictor: " << FLAGS_modular_predictor
-                  << std::endl;
-        return EXIT_FAILURE;
+
+      process_flag("center_x", FLAGS_center_x,
+                   JXL_ENC_FRAME_SETTING_GROUP_ORDER_CENTER_X,
+                   [](int32_t x) -> std::string { return ""; });
```

Comment:
I guess we could check if `-1 <= x` and return "valid values are -1 and [0, xsize)" or something similar if this is not the case.